### DISCUSSION TITLE
How to exclude duration of on_task_start() or user on_start() from runtime duration

### DISCUSSION THREAD

**Discussion by sfc-gh-tlasica** [locustio/locust#2884] [Q&A]
I have a benchmark that I would like to run for 5 minutes.
There are some event handlers such as `on_task_start()` or user `on_start()` that take non-trivial e.g. 30s.
If I set runtime to 60s then users will only use 30s for running tasks.
I would like my users to run tasks for 60s (so do not include preparation time).

Is there a way to reset timer in `on_test_start()` event?

**Comment by cyberw** [comment]
Hi! There's no simple way to do that at the moment. you can call `runner.quit()` "manually" when you want though.
